mongo/options/autoencryptionoptions.go

```diff
@@ -32,13 +35,16 @@ type AutoEncryptionOptions struct {
 	BypassAutoEncryption  *bool
 	ExtraOptions          map[string]interface{}
 	TLSConfig             map[string]*tls.Config
+	HTTPClient            *http.Client
```

Comment:
Optional: Consider adding a comment that describes what the `HTTPClient` configuration is used for.

E.g.
```go
// HTTPClient allows specifying a custom HTTP client used to fetch credentials from local
// credentials services (e.g. the EC2 instance metadata service). If nil, then the Go Driver
// uses its default HTTP client (internal.DefaultHTTPClient).
```